src/Adapter/Product/CommandHandler/DeleteProductHandler.php

```diff
@@ -47,18 +44,28 @@ class DeleteProductHandler implements DeleteProductHandlerInterface
     private $productRepository;
 
     /**
-     * @param ProductRepository $productRepository
+     * @var ProductDeleter
      */
-    public function __construct(ProductRepository $productRepository)
-    {
+    private $productDeleter;
+
+    public function __construct(
+        ProductRepository $productRepository,
+        ProductDeleter $productDeleter
+    ) {
         $this->productRepository = $productRepository;
+        $this->productDeleter = $productDeleter;
     }
 
     /**
      * {@inheritdoc}
      */
     public function handle(DeleteProductCommand $command): void
     {
-        $this->productRepository->deleteByShopConstraint($command->getProductId(), ShopConstraint::allShops());
+        $productId = $command->getProductId();
+
+        $this->productDeleter->deleteFromShops(
```

Comment:
Same suggestion as in bulk handler
```suggestion
        $this->productDeleter->deleteByShopConstraint(
```